xblock/fields.py

```diff
@@ -646,6 +650,24 @@ def is_set_on(self, xblock):
 
     def __hash__(self):
         return hash(self.name)
+    
+    @classmethod
+    def Query(cls, field_name):
+        """
+        Return a instance of Query which allow ad-hoc queries to access shared fields. 
+        """
+        return Query(field_name)
```

Comment:
Yes, I agree that Query can just be an XBlock service, instead of an inner
class of Field. I will see how far I can proceed on this path.

On Thu, Aug 13, 2015 at 4:15 PM, Calen Pennington <notifications@github.com>
wrote:

> In xblock/fields.py
> <https://github.com/edx/XBlock/pull/317#discussion_r37019946>:
>
> > @@ -646,6 +650,24 @@ def is_set_on(self, xblock):
> >
> >      def __hash__(self):
> >          return hash(self.name)
> > +
> > +    @classmethod
> > +    def Query(cls, field_name):
> > +        """
> > +        Return a instance of Query which allow ad-hoc queries to access shared fields.
> > +        """
> > +        return Query(field_name)
>
> re field_name: Pushing it further, couldn't you just pass the field_name
> at the time that you execute the query? In which case, maybe Query should
> just be an XBlock service, and Shared is the only field-like thing.
>
> Alternately, you could allow the user to pass a whole set of things to the
> Query object, to pre-seed (and allow the app to pre-load) as much as
> possible. So, if you gave it a block_type and a field_name, then when you
> query, it can check that the usage you passed is the block type you were
> expecting (for instance).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/XBlock/pull/317/files#r37019946>.
>



-- 
Sincerely,
Xiaolu Xiong
http://www.wpi.edu/~xxiong/